docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,528 @@
+
+## Create a custom modular resource
+
+To create your own modular resource, code a module in Go or Python using the module support libraries provided by [Viam's SDKs](/program/apis/) that implements at least one new model or {{< glossary_tooltip term_id="api-namespace-triplet" text="subtype" >}} of {{< glossary_tooltip term_id="resource" text="resource" >}}:
+
+{{< tabs >}}
+{{% tab name="New Model" %}}
+Define a new model of a built-in resource subtype:
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (ex. `rdk:component:base`).
+Import your custom model and API into the main program and register the new resource model with your chosen SDK's helper methods.
+
+1. [Code a main program](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+1. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
+
+{{% /tab %}}
+{{% tab name="New Type" %}}
+Define a new {{< glossary_tooltip term_id="api-namespace-triplet" text=" type or subtype" >}} of resource:
+
+1. Define the messages and methods of the new API in [protobuf](https://github.com/protocolbuffers/protobuf), then [generate code](https://grpc.io/docs/languages/python/generated-code/) in Python or Go and use the generated code to implement the higher level server and client functions required.
+
+2. [Code at least one model](#code-a-new-resource-model) of this new resource.
+Make sure to implement every method required in your API definition.
+Import your custom models and APIs into the main program and register the new resource models with your chosen SDK's helper methods.
+
+1. [Code a main program](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+1. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
+
+{{% /tab %}}
+{{% /tabs %}}
+
+### Code a new resource model
+
+The following example module registers a modular resource implementing Viam's built-in [Base API](/components/base/#api) [(rdk:service:base)](/extend/modular-resources/key-concepts/#models) as a new model, `"mybase"`:
+
+The Go code for the custom model (<file>mybase.go</file>) and module entry point file (<file>main.go</file>) are adapted from the full demo modules available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources).
+
+{{< tabs name="Sample SDK Code">}}
+{{% tab name="Go"%}}
```

Comment:
Is there a reason for showing Go before Python on all of these, contrary to what we do in the rest of the docs?

Reply:
No just switched up in my head, fixed!